src/test/java/bio/terra/service/snapshotbuilder/SnapshotBuilderServiceTest.java

```diff
@@ -67,4 +84,28 @@ void enumerateSnapshotRequestsByDatasetId() {
         snapshotBuilderService.enumerateByDatasetId(datasetId),
         equalTo(expected));
   }
+
+  @ParameterizedTest
+  @EnumSource(CloudPlatform.class)
+  public void getConceptChildren(CloudPlatform cloudPlatform) {
+    Dataset dataset =
+        new Dataset(new DatasetSummary().cloudPlatform(cloudPlatform))
+            .projectResource(new GoogleProjectResource().googleProjectId("project123"))
+            .name("dataset123")
+            .id(UUID.randomUUID());
+    when(datasetService.retrieve(dataset.getId())).thenReturn(dataset);
+    CloudPlatformWrapper cloudPlatformWrapper = CloudPlatformWrapper.of(cloudPlatform);
+    if (cloudPlatformWrapper.isGcp()) {
+      when(bigQueryDatasetPdao.runQuery(any(), any(), any()))
+          .thenReturn(List.of(new SnapshotBuilderConcept().name("concept1").id(1)));
+    } else {
+      when(azureSynapsePdao.runQuery(any(), any()))
+          .thenReturn(List.of(new SnapshotBuilderConcept().name("concept1").id(1)));
+    }
+    var response = snapshotBuilderService.getConceptChildren(dataset.getId(), 1, null);
+    assertThat(
+        "getConceptChildren returns the expected response",
+        response.getResult().get(0).getName(),
+        equalTo("concept1"));
```

Comment:
Would it work to assert on the object itself, something like
```
    var concepts = List.of(new SnapshotBuilderConcept().name("concept1").id(1));
    ...
    if (cloudPlatformWrapper.isGcp()) {
      when(bigQueryDatasetPdao.runQuery(any(), any(), any())).thenReturn(concepts);
    } else {
      when(azureSynapsePdao.runQuery(any(), any())).thenReturn(concepts);
    }
    ...
    assertThat(
        "getConceptChildren returns the expected response",
        response,
        equalTo(new SnapshotBuilderGetConceptsResponse().result(concepts)));
```

Reply:
That's a great point!